.github/workflows/xahau-ga-nix.yml

```diff
@@ -238,7 +248,7 @@ jobs:
                   echo "Hiding GCC $version -> renaming to $counter (will be seen as GCC version $counter)"
                   # Safety check: ensure target doesn't already exist
                   if [ ! -e "/usr/lib/gcc/x86_64-linux-gnu/$counter" ]; then
-                    sudo mv "$dir" "/usr/lib/gcc/x86_64-linux-gnu/$counter"
+                    mv "$dir" "/usr/lib/gcc/x86_64-linux-gnu/$counter"
```

Comment:
Via presence of cached_enabled (gha_cache_enabled) we seem to want to support both self hosted runners and github actions runners, yet removing sudo here will cause us some issues for the latter.

Reply:
Yeah, I just realized that before. I was about to ask "why not just simplify it and remove that option", but MacOS